GSOC18/GeometryUtilities-master/interpolation/hexCells_to_squareCell_interpolation.py

```diff
@@ -51,15 +56,23 @@ def linear_interpolate_hex_to_square(hex_cells_dict,layer,resolution=(500,500)):
 
     #Iterating over all the cells to get the bounds of the detector
     print '>>> Calculating Bounds'
-    center_x=map(lambda c:c.center.x,cells_dict.values())
-    max_x=max(center_x)
-    min_x=min(center_x)
-    center_y=map(lambda c:c.center.y,cells_dict.values())
-    max_y=max(center_y)
-    min_y=min(center_y)
+    bounds_cords=map(lambda c:c.vertices.bounds,cells_dict.values())
+    max_x=max(cords[2] for cords in bounds_cords)
+    min_x=min(cords[0] for cords in bounds_cords)
+    max_y=max(cords[3] for cords in bounds_cords)
+    min_y=min(cords[1] for cords in bounds_cords)
     t2=datetime.datetime.now()
+    print 'Bounds: xmin %s ,xmax %s ,ymin %s ,ymax %s '%(min_x,max_x,
+                                                        min_y,max_y)
     print 'Bounding completed in: ',t2-t1,' sec\n'
 
+    #Calculating the Resolution (based on exp_edge_length)
```

Comment:
These changes are for the calculation of the resolution from the expected edge length of the square cell.